adserver/analyzer/tasks.py

```diff
@@ -47,36 +53,64 @@ def analyze_url(url, publisher_slug):
 
 
 @app.task
-def daily_visited_urls(days=2):
-    """Analyze URLs that were visited recently but have never been analyzed."""
-    dt_cutoff = timezone.now() - datetime.timedelta(days=days)
-
-    # Loop over each publisher and query the offers table for distinct URLs
-    for publisher in Publisher.objects.all():
-        for obj in (
-            Offer.objects.filter(publisher=publisher, date__gt=dt_cutoff)
-            .values("url")
-            .distinct()
-        ):
-            url = obj["url"]
-            if not url:
-                # Blank URLs can't be analyzed
-                continue
-
-            normalized_url = normalize_url(url)
-            if not AnalyzedUrl.objects.filter(
-                url=normalized_url, publisher_id=publisher.id
-            ).exists():
-                analyze_url.apply_async(args=[normalized_url, publisher.slug])
+def daily_visited_urls_aggregation(day=None):
+    """Aggregate URLs and update how many times they were visited in the last day."""
+    if day is None:
+        # Do the previous day's reports
+        day = timezone.now() - datetime.timedelta(days=1)
+
+    start_date, end_date = get_day(day)
+
+    # Query the offers table for distinct URLs
+    for obj in (
+        Offer.objects.using(settings.REPLICA_SLUG)
+        .filter(
+            date__gte=start_date,
+            date__lt=end_date,
+        )
+        .filter(viewed=True)
+        .filter(advertisement__flight__campaign__campaign_type=PAID)
+        .values("url", "publisher_id")
+        .annotate(visits=models.Count("url"))
+        .iterator()
+    ):
+        url = obj["url"]
+        publisher_id = obj["publisher_id"]
+        visits = obj["visits"]
+
+        if not url:
+            # Blank URLs can't be analyzed
+            continue
+
+        normalized_url = normalize_url(url)
+
+        url_obj, created = AnalyzedUrl.objects.get_or_create(
+            url=normalized_url,
+            publisher_id=publisher_id,
+            defaults={
+                "last_ad_served_date": timezone.now(),
+                "visits_since_last_analyzed": visits,
+            },
+        )
+
+        if not created:
+            # If the object wasn't created, add the visits and set the ad last served date
+            url_obj.update(
+                last_ad_served_date=timezone.now(),
+                visits_since_last_analyzed=models.F("visits_since_last_analyzed")
+                + visits,
+            )
 
 
 @app.task
-def weekly_analyze_urls(days=7):
-    """Analyze URLs that have been visited but not analyzed since DAYS number of days."""
+def daily_analyze_urls(days=7, min_visits=50):
+    """Analyze URLs that have been visited VISITS times but not analyzed since DAYS number of days."""
```

Comment:
Should the kwarg be `visits` to match this docstring? The caps are a bit odd, I'd probably do:

```suggestion
    """Analyze URLs that have been visited `visits` times but not analyzed since `days` number of days."""
```

Reply:
I don't feel that strongly. I do think `min_visits` is a more clear variable name. Maybe just use `min_visits` everywhere?

Reply:
`min_visits` is fine, mostly just commenting on the docstring syntax.

Reply:
Fixed up.